### PR TITLE
:sparkles: add support for USMailV2, BillOfLadingV1, FR EnergyBillV1, FR Payslip V1 & NutritionFactsLabel V1

### PR DESCRIPTION
**Pull request by sebastianMindee**
## Description

Add support for:
* :sparkles: US Mail V2
* :sparkles: Bill of Lading V1
* :sparkles: FR Energy Bill V1
* :sparkles: FR Payslips V1
* :sparkles: Nutrition Facts Label V1

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Requires a change to the official Guide documentation.

### CODE REVIEW

File: src/Parsing/Common/SummaryHelper.php [hunk 1]
```diff
@@ -8,19 +8,27 @@
...
+            return '';
+        }
+        $formatted = number_format($number, 5, '.', '');
+
+        $formatted = rtrim(rtrim($formatted, '0'), '.');
+
+        if (substr_count($formatted, '.') == 0) {
+            $formatted .= '.00';
+        } elseif (strlen($formatted) - strpos($formatted, '.') <= 2) {
+            $formatted = sprintf("%.2f", $formatted);
```
Comment (ianardee) [inline src/Parsing/Common/SummaryHelper.php:28]: I'm so sorry

File: src/Parsing/Common/SummaryHelper.php [hunk 2]
```diff
@@ -49,13 +73,14 @@ public static function formatForDisplay($inputString = null, ?int $maxColSize =
...
-        if (!$inputString || strlen($inputString) == 0) {
+        $inputString = SummaryHelper::escapeSpecialChars($inputString);
+        if (!$inputString || mb_strlen($inputString) == 0) {
             return "";
         }
         if (!isset($maxColSize)) {
             return $inputString;
         }
-        return strlen($inputString) <= $maxColSize ? $inputString : substr(
+        return mb_strlen($inputString) <= $maxColSize ? $inputString : mb_substr(
```
Comment (ianardee) [inline src/Parsing/Common/SummaryHelper.php:83]: 1. Ternary expressions are harder to read, prefer more explicit if/else, even if longer.
2. Set encoding explicitly on mb_xxx functions.

File: src/Product/FinancialDocument/FinancialDocumentV1LineItem.php [hunk 3]
```diff
@@ -100,16 +114,16 @@ private function printableValues(): array
...
-        $outStr .= str_pad($printable["unitMeasure"], 15) . " | ";
-        $outStr .= str_pad($printable["unitPrice"], 10) . " | ";
+        $outStr .= mb_str_pad($printable["description"], 36, ' ', STR_PAD_RIGHT, "UTF-8") . " | ";
+        $outStr .= mb_str_pad($printable["productCode"], 12, ' ', STR_PAD_RIGHT, "UTF-8") . " | ";
+        $outStr .= mb_str_pad($printable["quantity"], 8, ' ', STR_PAD_RIGHT, "UTF-8") . " | ";
+        $outStr .= mb_str_pad($printable["taxAmount"], 10, ' ', STR_PAD_RIGHT, "UTF-8") . " | ";
+        $outStr .= mb_str_pad($printable["taxRate"], 12, ' ', STR_PAD_RIGHT, "UTF-8") . " | ";
+        $outStr .= mb_str_pad($printable["totalAmount"], 12, ' ', STR_PAD_RIGHT, "UTF-8") . " | ";
+        $outStr .= mb_str_pad($printable["unitMeasure"], 15, ' ', STR_PAD_RIGHT, "UTF-8") . " | ";
+        $outStr .= mb_str_pad($printable["unitPrice"], 10, ' ', STR_PAD_RIGHT, "UTF-8") . " | ";
```
Comment (ianardee) [inline src/Product/FinancialDocument/FinancialDocumentV1LineItem.php:126]: Lots of duplicate code, I would probably create a static method in `SummaryHelper` (called for each field), but up to you.

File: src/Product/BillOfLading/BillOfLadingV1.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    public function __construct(array $rawPrediction)
+    {
+        parent::__construct($rawPrediction);
+        $this->prediction = new BillOfLadingV1Document($rawPrediction['prediction']);
+        $this->pages = [];
+        foreach ($rawPrediction['pages'] as $page) {
+            try {
+                $this->pages[] = new Page(BillOfLadingV1Document::class, $page);
+            } catch (MindeeUnsetException $ignored) {
+            }
```
Comment (ianardee) [inline src/Product/BillOfLading/BillOfLadingV1.php:37]: Maybe add a debug log here, no? Just an idea.

Not sure how common this is, if it happens systematically then not worth logging.

**Review by ianardee — CHANGES_REQUESTED** [review]

File: src/Parsing/Common/SummaryHelper.php (hunk 2 above)
Comment (sebastianMindee) [reply src/Parsing/Common/SummaryHelper.php:83]: I KNEW I was going to miss one :angry:

Comment (ianardee) [reply src/Parsing/Common/SummaryHelper.php:83]: See point n° 1 ;-)

File: src/Parsing/Common/SummaryHelper.php (hunk 1 above)
Comment (sebastianMindee) [reply src/Parsing/Common/SummaryHelper.php:28]: It's truly fascinating how blunt of a tool PHP is at times

**Review by ianardee — APPROVED** [review]

**Pull request merged**